### PR TITLE
Edit task ratings

### PR DESCRIPTION
**Pull request by IidaKainu**
Added RoadmapReadUsers permission to Business-role
- Business-users can read either Developer-users or users with at least one same represented customer
- This enables business-users to see who rated listed ratings. It could have been confusing otherwise, as the user could not even identify their own ratings in the list

### CODE REVIEW

**Comment by IidaKainu** [comment]
Moved edit icons to the right side of an editable rating

**Review by juhoeerola — CHANGES_REQUESTED** [review]
Seems to work as expected

**Review by PatricKangasmaki — COMMENTED** [review]
Works well, good job 👍 

**Shower thoughts about the design:** One thing that caught my eye was that each of the users task have their own edit buttons even though they all open the same modal where there are all the tasks. The design indicated (at least to me) that I would just change the rating of that exact task. If there was, lets say 100, ratings, it could be frustrating to scroll though all those in the modal just to find the one that you wanted to edit. On the other hand, editing all at the same time is needed if you actually wanted to change possibly tens of ratings at the same time.

image

File: frontend/src/pages/TaskOverviewPage.tsx [hunk 1]
```diff
@@ -129,16 +129,22 @@ const TaskOverview: FC<{
         onDataEditConfirm={handleEditConfirm}
         key={task.id}
       />
-      {hasPermission(role, Permission.RoadmapReadUsers) && (
```
[the remark this answers is not in the document]
Comment (IidaKainu) [reply frontend/src/pages/TaskOverviewPage.tsx]: > I'm not quite sure all roles having the permission at the moment justifies this.

Yeah now that it is pointed out, the permission check should be left as it is

**Comment by IidaKainu** [comment]
Added Juho's suggestions 👍 

> they all open the same modal where there are all the tasks

In last weekly we discussed it can be like this, at least for now

> The design indicated (at least to me) that I would just change the rating of that exact task

I do still agree with this. If we want to change the modal to display only a certain task on edit, I'd add it as another task to do: it requires some logic changes to the modal, which could take a while.

**Review by PatricKangasmaki — APPROVED** [review]

**Review by juhoeerola — APPROVED** [review]

**Pull request merged**